packages/frames.js/src/getAddressForFid.ts

```diff
@@ -40,33 +40,28 @@ export async function getAddressForFid<
       throw new Error(
         `Failed to parse response body as JSON because server hub returned response with status "${response.status}" and body "${await response.clone().text()}"`
       );
-    })) as { messages?: Message[] };
+    })) as { messages?: Record<string, any>[] };
 
-  if (
-    messages?.[0]?.data &&
-    messages[0].data.type === MessageType.VERIFICATION_ADD_ETH_ADDRESS
-  ) {
-    const {
-      data: {
-        // @ts-expect-error this is correct but somehow it is missing in Message definition
-        verificationAddEthAddressBody: { address },
-      },
-    } = messages[0];
-    return address;
-  } else if (fallbackToCustodyAddress) {
-    const publicClient = createPublicClient({
-      transport: http(),
-      chain: optimism,
-    });
-    const address = await publicClient.readContract({
-      abi: parseAbi(["function custodyOf(uint256 fid) view returns (address)"]),
-      // IdRegistry contract address
-      address: "0x00000000fc6c5f01fc30151999387bb99a9f489b",
-      functionName: "custodyOf",
-      args: [BigInt(fid)],
-    });
-    return address;
+  if (!messages || messages.length === 0) {
```

Comment:
does this consider what happens if a message is a solana verification and not the correct type?

Reply:
I understand what you mean, not sure how many things can `verificationsByFid` return but the message types should be only verifications right?

Reply:
Ok now I'm looking for first verified address, otherwise we return `null`.